src/editor/nodes/ParticleNode.js

```diff
@@ -0,0 +1,416 @@
+	      c * (gl_PointCoord.y - 0.5) - s * (gl_PointCoord.x - 0.5) + 0.5);  // rotate UV coordinates to rotate texture
+    	  vec4 rotatedTexture = texture2D( texture,  rotatedUV );
+        gl_FragColor = gl_FragColor * rotatedTexture;
+        //gl_FragColor = gl_FragColor * texture2D( texture, gl_PointCoord );
+        
+        if ( vPosZ < -0.1) discard;
+        
+				
+			}
+  `;
+
+export default class ParticleNode extends EditorNodeMixin(THREE.Points) {
+  static legacyComponentName = "particle";
+
+  static nodeName = "Particle";
+
+  static initialElementProps = {
+    src: new URL(defaultParticleUrl, location).href
+  };
+
+  static async deserialize(editor, json, loadAsync) {
+    const node = await super.deserialize(editor, json);
+
+    const {
+      src,
+      prewarm,
+      colorCurve,
+      velocityCurve,
+      startColor,
+      middleColor,
+      endColor,
+      startOpacity,
+      middleOpacity,
+      endOpacity,
+      emitterWidth,
+      emitterHeight,
+      size,
+      velocity,
+      endVelocity,
+      angularVelocity,
+      particleCount,
+      lifetime,
+      lifetimeRandomness
+    } = json.components.find(c => c.name === "particle").props;
+
+    node.startColor.set(startColor);
+    node.middleColor.set(middleColor);
+    node.endColor.set(endColor);
+    node.colorCurve = colorCurve;
+    node.velocityCurve = velocityCurve;
+    node.prewarm = prewarm;
+    node.startOpacity = startOpacity || 1;
+    node.middleOpacity = middleOpacity || 1;
+    node.endOpacity = endOpacity || 1;
+    node.emitterHeight = emitterHeight || 1;
+    node.emitterWidth = emitterWidth || 1;
+    node.lifetime = lifetime || 5;
+    node.size = size || 1;
+    node.lifetimeRandomness = lifetimeRandomness || 5;
+    node.particleCount = particleCount || 1000;
+    node.velocity.copy(velocity);
+    node.angularVelocity = angularVelocity || 0;
+    node.endVelocity = endVelocity || new THREE.Vector3(0, 0, 0.5);
+
+    loadAsync(
+      (async () => {
+        await node.load(src);
+      })()
+    );
+    node.createParticle();
+
+    return node;
+  }
+
+  static async load() {
+    defaultParticleSprite = await new Promise((resolve, reject) => {
+      new THREE.TextureLoader().load(defaultParticleUrl, resolve, null, e =>
+        reject(`Error loading Image. ${eventToMessage(e)}`)
+      );
+    });
+    defaultParticleSprite.flipY = false;
+  }
+
+  constructor(editor) {
+    const geometry = new THREE.BufferGeometry();
+    const material = new THREE.RawShaderMaterial({
+      uniforms: {
+        texture: { value: defaultParticleSprite }
+      },
+      vertexShader,
+      fragmentShader,
+      transparent: true,
+      depthWrite: false,
+      blendEquation: THREE.AdditiveBlending
+    });
+
+    super(editor, geometry, material);
+
+    this.lastUpdated = 0;
+    this._canonicalUrl = "";
+    this.emitterHeight = 1;
+    this.emitterWidth = 1;
+    this.initialPositions = [];
+    this.size = 1;
```

Comment:
The default particle size seems a little big right now. Maybe `0.25` is a better value?